src/Controls/src/Core/Shell/ShellNavigationManager.cs

```diff
@@ -115,7 +115,11 @@ public ShellNavigationManager(Shell shell)
 				modalStackPreBuilt = true;
 
 				bool? isAnimated = (nextActiveSection != currentShellSection) ? false : animate;
-				await nextActiveSection.GoToAsync(navigationRequest, parameters, _shell.FindMauiContext()?.Services, isAnimated, isRelativePopping);
+
+				await _shell.Dispatcher.DispatchAsync(() =>
```

Comment:
I'm curious to investigate this a bit further and see if we can base this fix on something more deterministic.